### ISSUE TITLE
Very suspicious code in TMRpcm.cpp

### ISSUE THREAD

**Issue by thegoodhen** [TMRh20/TMRpcm#100]
if (!card1.erase(bgnBlock, endBlock)) //Serial.println("card.erase failed");

	rut.close();
        fil.close();

On lines 1631-1634 in TMRpcm.cpp

This makes the "rut" close only when card1.erase fails. Is this the intentional behavior?

**Comment by TMRh20** [comment]
I think so, it works, so it must be intentional.

**Comment by thegoodhen** [comment]
Let me rephrase that. You have a bug in your code. I am pretty sure it wasn't intentional.

**Comment by TMRh20** [comment]
Lol I guess I’ll need to hook up some stuff and test this out. It will take some time.

**Comment by thegoodhen** [comment]
Well I can confirm it works when I change

if (!card1.erase(bgnBlock, endBlock)) //Serial.println("card.erase failed");

to

if (!card1.erase(bgnBlock, endBlock)) Serial.println("card.erase failed");

(Which effectively causes the rut.close() to run no matter what).

**Comment by TMRh20** [comment]
Fixed, thanks.

**Comment by TMRh20** [comment]
Awesome, I’ll trust your testing and make the changes.
